pkg/types/chains/evm/evm.go

```diff
@@ -108,3 +114,50 @@ type Head struct {
 type TransactionFee struct {
 	TransactionFee *big.Int // Cost of transaction in wei
 }
+
+type SignedReport struct {
+	RawReport     []byte
+	ReportContext []byte
+	Signatures    [][]byte
+	ID            []byte
+}
+
+// TransactionStatus is the result of the transaction sent to the chain
+type TransactionStatus int
+
+const (
+	//Transaction was sent successfully to the chain and successfully executed
+	TxSuccess TransactionStatus = iota
+	//Transaction was sent successfully to the chain but the smart contract execution reverted
+	TxReverted
+	//Transaction was not sent successfully to the chain
+	TxFatal
+)
+
+type TxError struct {
+	// Internal ID used for tracking purposes of transactions.
+	TxID string
+}
+
+func (e *TxError) Error() string {
+	return fmt.Sprintf("Fail processing Transaction with internal TxID: %s", e.TxID)
+}
+
+type TransactionResult struct {
+	TxStatus TransactionStatus
+	TxHash   Hash
```

Comment:
The idea of a tx result containing a tx hash doesn't fit my mental model. Normally a tx hash is available before even sending, and the result or status has to do with inclusion. In this case the caller can't produce the hash since we construct the transaction, so I understand that we need to return it, but is this payload more like just a `Transaction` rather than a `Result` per se?
```suggestion
type Transaction struct {
	Status TransactionStatus
	Hash   Hash
```

Reply:
A quick follow up is fine if it's not affecting anyone yet, but keeping this API stable is important to reduce friction.

Reply:
> FWIW geth JSON and GO clients return transaction types with extra fields set conditionally depending on status, so there is precedent for that 

Just want to +1 this, we should keep this familiar with geth ux